### PR TITLE
Add actionlint to diagnostics builtins

### PR DESCRIPTION
**Pull request by mattleong**
Add [actionlint](https://github.com/rhysd/actionlint) to diagnostic builtins

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics/actionlint.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+local DIAGNOSTICS = methods.internal.DIAGNOSTICS
+
+return h.make_builtin({
+    name = "actionlint",
+    method = DIAGNOSTICS,
+    filetypes = { "yaml" },
+    generator_opts = {
+        command = "actionlint",
+        args = { "-no-color", "-format", "{{json .}}", "-" },
+        format = "json_raw",
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/actionlint.lua:13]: Is it sufficient to use `json` here? We generally only want `json_raw` if we plan on handling errors ourselves.

File: lua/null-ls/builtins/diagnostics/actionlint.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        from_stderr = true,
+        to_stdin = true,
+        on_output = function(params)
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/actionlint.lua:16]: We can do this a bit more directly:

```lua
on_output = h.diagnostics.from_json({
    attributes = {
        message = "message",
        source = "actionlint",
        code = "kind",
        severity = 1,
    },
})
```

**Review by jose-elias-alvarez — COMMENTED** [review]
Left a couple of comments but this looks useful. Thanks!

File: lua/null-ls/builtins/diagnostics/actionlint.lua (hunk 1 above)
Comment (mattleong) [reply lua/null-ls/builtins/diagnostics/actionlint.lua]: Yup! `json` works just fine

**Comment by mattleong** [comment]
@jose-elias-alvarez Updated 👍

**Comment by jose-elias-alvarez** [comment]
Thanks!

**Pull request merged**